### PR TITLE
Enable some linters

### PR DESCRIPTION
**Pull request by qwerty287**
Mostly those that did not require much work.

From #2960

### CODE REVIEW

File: server/pipeline/topic.go [hunk 1]
```diff
@@ -33,9 +34,14 @@ func publishToTopic(pipeline *model.Pipeline, repo *model.Repo) {
 	}
 	pipelineCopy := *pipeline
 
-	message.Data, _ = json.Marshal(model.Event{
+	var err error
+	message.Data, err = json.Marshal(model.Event{
```
Comment (anbraten) [inline server/pipeline/topic.go:38]:
```suggestion
	message.Data, err := json.Marshal(model.Event{
```

Should work as well?

File: server/pipeline/queue.go [hunk 2]
```diff
@@ -42,11 +42,15 @@ func queuePipeline(repo *model.Repo, pipelineItems []*stepbuilder.Item) error {
 		task.RunOn = item.RunsOn
 		task.DepStatus = make(map[string]model.StatusValue)
 
-		task.Data, _ = json.Marshal(rpc.Workflow{
+		var err error
+		task.Data, err = json.Marshal(rpc.Workflow{
```
Comment (anbraten) [inline server/pipeline/queue.go:46]:
```suggestion
		task.Data, err := json.Marshal(rpc.Workflow{
```

File: pipeline/frontend/metadata/environment.go [hunk 3]
```diff
@@ -136,7 +138,10 @@ func (m *Metadata) Environ() map[string]string {
 		params["CI_PIPELINE_FILES"] = "[]"
 	} else if len(m.Curr.Commit.ChangedFiles) <= maxChangedFiles {
 		// we have to use json, as other separators like ;, or space are valid filename chars
-		changedFiles, _ := json.Marshal(m.Curr.Commit.ChangedFiles)
+		changedFiles, err := json.Marshal(m.Curr.Commit.ChangedFiles)
+		if err != nil {
+			log.Error().Err(err).Msg("marshal changed files")
```
Comment (anbraten) [inline pipeline/frontend/metadata/environment.go:143]:
```suggestion
      // TODO: forward error to user
			log.Error().Err(err).Msg("marshal changed files")
```

File: cmd/agent/agent.go [hunk 4]
```diff
@@ -271,6 +271,8 @@ func getBackendEngine(backendCtx context.Context, backendName string, addons []s
 }
 
 func runWithRetry(context *cli.Context) error {
+	initHealth()
```
Comment (anbraten) [inline cmd/agent/agent.go:274]: Was it previously only used by importing the package?

File: agent/state.go [hunk 5]
```diff
@@ -74,7 +74,10 @@ func (s *State) Healthy() bool {
 
 func (s *State) WriteTo(w io.Writer) (int64, error) {
 	s.Lock()
-	out, _ := json.Marshal(s)
+	out, err := json.Marshal(s)
+	if err != nil {
+		return 0, err
```
Comment (6543) [inline agent/state.go:79]: s.Unloik

**Review by xoxys — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**